### PR TITLE
Feature: opt out user metrics

### PR DESCRIPTION
**Pull request by RedTo**
Show info dialog at first start.
Enable analytics by default.

### CODE REVIEW

**Comment by EduardBartolovic** [comment]
Just a Note: this could make problems with the ePrivacy reform which will come next year

**Comment by RedTo** [comment]
Please go more in detail...

EduardBartolovic <notifications@github.com> schrieb am Di., 26. Juni 2018,
12:10:

**Comment by RedTo** [comment]
Okay I see. Thanks for this information. I just thought that there would be no user which will allow us to track any data if we use the opt-in mechanism...

But okay, than it seems that we have to leave it as opt-in...

**Comment by EduardBartolovic** [comment]
I think if you display the option in the beginning then there will be some people who will use it.

**Comment by AndiAndreas** [comment]
Opt-in with option to toggle it on first start is a cool way i think.

**Comment by RedTo** [comment]
Yeah, I just thought about that at the moment...

**Comment by RedTo** [comment]
Do you think a simple dialog with the option ok and no thanks (or something similar) is enough?

**Comment by EduardBartolovic** [comment]
This would be perfect

**Comment by AndiAndreas** [comment]
Sure

**Comment by RedTo** [comment]
image

There is an dialog where the user can decide to share anonymous user data or not. This dialog is opend every time the login page is visited. (I have done this as of the following scenario: User A says, it's ok to collect data and use the app. Then User B want to check something on the phone of A and login without being asked if he want to share or not, because the dialog will just come up once at the first app start. So I think poping up before every login (normally there are not that often changes of users I guess) is ok)

Sorry for the long sentences :-/

File: lib/views/settings.dart [hunk 1]
```diff
@@ -25,7 +27,7 @@ class _SettingsState extends State<Settings> {
   final CurrentUserPresenter currentUserPresenter;
   int credits = 0;
   int engCredits = 0;
-  bool isMetricsEnabled = false;
```
Comment (AndiAndreas) [inline lib/views/settings.dart]: Why true? What If popup fails? Should rather be false by default and then be set.

Comment (RedTo) [reply lib/views/settings.dart]: That's not really a default value. Just the initialize, but I have changed it...

A few lines later it is overwritten each time with the user settings and those can't be null at this time. But you are right - we never now...

**Pull request merged**